### PR TITLE
Allow to configure if IOSEQ_ASYNC is used per EventLoopGroup

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

There may be situations when the user dont want to use IOSEQ_ASYNC so we
should allow to configure this

Modifications:

Make it configurable if IOSEQ_ASYNC should be used

Result:

More flexible configuration

### CODE REVIEW

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java [hunk 1]
```diff
@@ -56,7 +56,7 @@
     private long prevDeadlineNanos = NONE;
     private boolean pendingWakeup;
 
-    IOUringEventLoop(IOUringEventLoopGroup parent, Executor executor, int ringSize,
```
Comment (hyperxpro) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java:59]: How about adding little bit of Documentation or Reference so user can understand what exactly it is. Maybe 'man page' link?

**Review by hyperxpro — COMMENTED** [review]
Just 1 comment

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java (hunk 1 above)
Comment (normanmaurer) [reply transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java:59]: done in `IOUringEventLoopGroup`

**Review by njhill — APPROVED** [review]
LGTM!

I also still wonder whether it might make sense to set this selectively depending on the op.. maybe something to investigate in future

**Comment by normanmaurer** [comment]
@njhill yep we can experiment with that as well... We could also make it configurable per `Channel` via a `ChannelOption` but I thought we should just keep it simple for now.

**Comment by 1Jo1** [comment]
> I also still wonder whether it might make sense to set this selectively depending on the op.. maybe something to investigate in future

@njhill yeah good point 👍 I think it doesn't make sense to use ASYNC flag for polling :)

**Pull request merged**